runtimes/go/appruntime/apisdk/cors/cors.go

```diff
@@ -67,8 +67,20 @@ func Options(cfg *config.CORS, staticAllowedHeaders, staticExposedHeaders []stri
 		AllowOriginRequestFunc: func(r *http.Request, origin string) bool {
 			// If the request has credentials, look up origins in AllowOriginsWithCredentials.
 			// Credentials are cookies, authorization headers, or TLS client certificates.
-			hasCreds := len(r.Cookies()) > 0 || r.Header["Authorization"] != nil || (r.TLS != nil && len(r.TLS.PeerCertificates) > 0)
-			if hasCreds {
+			hasCreds := func(r *http.Request) bool {
+				if len(r.Cookies()) > 0 || r.Header["Authorization"] != nil || (r.TLS != nil && len(r.TLS.PeerCertificates) > 0) {
+					return true
+				}
+
+				if r.Method == http.MethodOptions {
+					if val := r.Header.Get("Access-Control-Request-Headers"); val != "" {
+						return strings.Contains(strings.ToLower(val), "authorization")
```

Comment:
Should we check for `cookie` here too? Not sure it is present. Also, according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Request-Headers the header names are always lowercase, so can skip `strings.ToLower`.

Finally, we need to be more careful than just `strings.Contains`, otherwise someone could set a header like `x-dummy-authorization` and we would pick it up here. Possibly we could use `r.Header.Values` but I'm not sure

Reply:
`.Values` only returns the value for each header, if it is set multiple times, so we need to split as well